include/deal.II/matrix_free/mapping_info.templates.h

```diff
@@ -1100,6 +1134,185 @@ namespace internal
           }
       }
 
+
+
+      /**
+       * This evaluates the mapping information on a range of cells calling
+       * into the tensor product interpolators of the matrix-free framework,
+       * using a polynomial expansion of the cell geometry in terms of
+       * MappingQ.
+       */
+      template <int dim, typename Number, typename VectorizedArrayType>
+      void
+      compute_range_mapping_q(
+        const unsigned int               begin_cell,
+        const unsigned int               end_cell,
+        const std::vector<GeometryType> &cell_type,
+        const std::vector<bool> &        process_cell,
+        const UpdateFlags                update_flags_cells,
+        const AlignedVector<double> &    plain_quadrature_points,
+        const FE_DGQ<dim> &              fe_geometry,
+        MappingInfoStorage<dim, dim, Number, VectorizedArrayType> &my_data)
+      {
+        // we want to use vectorization for computing the quantities, but must
+        // evaluate the geometry in double precision; thus, for floats we need
+        // to do things in two sweeps and convert the final result
```

Comment:
Why are you doing this? Is mapping in deal.II always double precision?

Reply:
Yes, the geometry is always double precision. I actually started out with doing everything in the native `Number` type requested by the user but found out that a number of tests failed when done in `float`. I cannot say at this point if the accuracy is really important but given that we do this initialization once I think we can keep things more accurate this way. If we wanted to do things on the fly, we would of course compute everything from `float` representations of the mapping support points - that's the whole point in getting things more quickly than with precomputed arrays.